staging/src/k8s.io/component-base/metrics/testutil/metrics.go

```diff
@@ -192,23 +266,26 @@ func GetHistogramFromGatherer(gatherer metrics.Gatherer, metricName string) (His
 	}
 
 	if metricFamily == nil {
-		return Histogram{}, fmt.Errorf("metric %q not found", metricName)
+		return nil, fmt.Errorf("metric %q not found", metricName)
 	}
 
 	if metricFamily.GetMetric() == nil {
-		return Histogram{}, fmt.Errorf("metric %q is empty", metricName)
+		return nil, fmt.Errorf("metric %q is empty", metricName)
 	}
 
 	if len(metricFamily.GetMetric()) == 0 {
-		return Histogram{}, fmt.Errorf("metric %q is empty", metricName)
+		return nil, fmt.Errorf("metric %q is empty", metricName)
 	}
 
-	return Histogram{
-		// Histograms are stored under the first index (based on observation).
-		// Given there's only one histogram registered per each metric name, accessing
-		// the first index is sufficient.
-		metricFamily.GetMetric()[0].GetHistogram(),
-	}, nil
+	vec := make(HistogramVec, 0)
+	for _, metric := range metricFamily.GetMetric() {
+		if LabelsMatch(metric, lvMap) {
+			if hist := metric.GetHistogram(); hist != nil {
+				vec = append(vec, &Histogram{hist})
```

Comment:
This looks like is problematic when processing the metric like `scheduler_framework_extension_point_duration_seconds`, it will aggregate all the `Histogram` which belong to the family, but what we want is the individual `Histogram` for each extension point, aggregate "filter" and "score" etc. into one vector and do the calculation is meaningless IMHO. 

The issue could be workaround by defining `lvMap` for each of the extension point, but this is not efficient though, it will loop N times for N extension points.

There are two types of `HistogramVec`, something like `attempts` could be done in this way, but the extension points could not be aggregate in this way, we'd better to differentiate those two cases and come up a `HistogramVec` with labels.

Reply:
/lgtm 

let's tackle this in a follow up review.